tests/observers/slack_resource_test.cpp

```diff
@@ -0,0 +1,30 @@
+#include "gtest/gtest.h"
+
+#include "mesos/mesos.hpp"
+
+#include "observers/slack_resource.hpp"
+
+#include "tests/common/sources/json_source.hpp"
+
+#include "tests/common/sinks/dummy_sink.hpp"
+
+namespace mesos {
+namespace serenity {
+namespace tests {
+
+TEST(SlackResourceObserver, BasicTest) {
+  SlackResourceObserver observer;
+  JsonSource jsonSource;
+  DummySink<Resource> dummySink;
+
+  jsonSource.addConsumer(&observer);
+  observer.addConsumer(&dummySink);
+
+  jsonSource.RunTests("tests/fixtures/json_source_test.json");
+
+  ASSERT_EQ(dummySink.numberOfMessagesConsumed, 3);
+}
```

Comment:
It is really crucial to test it more preceisely. (: We can push this code, but we should test it properly soon - static analysis is not enough.